[CNA] Remove explicit `--turbopack` flag, when turbopack is enabled.

Nayeem-XTREME opened the pull request:
### What?

Remove explicit `--turbopack` flag, as turbopack is now default.

Closes NEXT-
Fixes #84216, #84282

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 473edb747d6e7978edc5ac5b5f5ce43d4f8c0ec3

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

Nayeem-XTREME committed:
refactor: update bundler flag handling in create-next-app templates

Removed the default `--turbopack` flag from the development script in package.json, as Turbopack is now the default bundler in v16+. The `--webpack` flag is only added when explicitly using Webpack. Updated related tests to reflect this change.

Nayeem-XTREME committed:
Merge branch 'canary' into remove-turbopack-flag-on-latest-version
